tests/unit/dataset/loader/test_random_pool.py

```
@@ -22,6 +22,27 @@
     """Tests for RandomPool model validation and functionality."""
 
+    def test_create_with_input_alias_for_text(self):
+        """RandomPool must map 'input' to 'text' for compatibility with OpenAI-style JSONL exports."""
+        data = RandomPool.model_validate(
+            {"type": "random_pool", "input": "What is machine learning?"}
+        )
+        assert data.text == "What is machine learning?"
+
+    def test_input_and_text_conflict_raises(self):
+        """Providing 'input' and 'text' with different values must raise a ValidationError."""
+        with pytest.raises(ValueError, match="input.*text"):
```

Ruff (0.15.10)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)